samples/commons/js/LinkedList.js

```diff
@@ -70,9 +79,57 @@ class LinkedList {
             first.next = null;
         }
         else this.head = null;
+        this.sizeOfList--;
         return first;
     }
 
+    insertAt(value, index) {
+        if (index > 0 && index > this.sizeOfList) {
```

Comment:
What if `index == size`? I guess this should be `index < 0 || size >= index`

Reply:
I think if `index === size`, then we can still insert at the end of it. 
But, if it is `index > size`, then we will not find a place to insert. 
But, I will check this one thoughtfully.